src/main.js

```diff
@@ -55,22 +59,30 @@ if (crashReporterDirectory) {
 		}
 	}
 
-	// Crashes are stored in the temp directory by default, so we
+	// Crashes are stored in the crashDumps directory by default, so we
 	// need to change that directory to the provided one
 	console.log(`Found --crash-reporter-directory argument. Setting temp directory to be '${crashReporterDirectory}'`);
-	app.setPath('temp', crashReporterDirectory);
-
-	// Start crash reporter
-	const { crashReporter } = require('electron');
-	const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
-	const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
-	crashReporter.start({
-		companyName: companyName,
-		productName: process.env['VSCODE_DEV'] ? `${productName} Dev` : productName,
-		submitURL: '',
-		uploadToServer: false
-	});
+	app.setPath('crashDumps', crashReporterDirectory);
+}
+
+// Start crash reporter for all processes
+const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
+const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
+const appCenter = product.appCenter;
+let submitURL = '';
+if (appCenter) {
+	const isWindows = (process.platform === 'win32');
+	const isMacintosh = (process.platform === 'darwin');
+	const isLinux = (process.platform === 'linux');
+	submitURL = isWindows ? appCenter[process.arch === 'ia32' ? 'win32-ia32' : 'win32-x64'] : isLinux ? appCenter[`linux-x64`] : appCenter.darwin;
+	//submitURL = submitURL.concat('&uid=', crashReporterId, '&iid=', crashReporterId, '&sid=', info.sessionId);
```

Comment:
how can we get `ITelemetryInfo.sessionId` here ?

Reply:
@deepak1556 not just that, but also the `crashReporterId`, no? My understanding is:
* `crashReporterId` is a unique number per installation (e.g. we store it in global storage only once)
* `sessionId` is a unique number that changes each time you start VSCode

The latter is probably used to be able to correlate a crash with telemetry session, but at least I never ever used this for investigating crashes. Maybe it would be sufficient to simply generate another UUID here if it is needed.

The crash reporter ID is more interesting... here we do not have access to the storage service yet. Maybe it could be written into `argv.json`? Since we already need to read that file on startup, we can reduce the perf hit this has.

Reply:
Thinking a bit more, encoding both these metadata in the url makes it difficult to have them on launch but there is way to provide additional metadata from each process even after launch using [crashReporter.addExtraParameter](https://github.com/electron/electron/blob/9-x-y/docs/api/crash-reporter.md#crashreporteraddextraparameterkey-value) , I just to need to refactor the appcenter implementation to recognize that.

Reply:
Only work left is on the backend, will create a PR today for it in the electron repo.